docs/v1-proposed-api.md

```diff
@@ -0,0 +1,242 @@
+  imageSet: # immutable
+    name: openshift-v4.1.18
+  images: {} # immutable
+  ingress:
+  - domain: apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+    name: default
+    servingCertificate: primary-cert-bundle
+  clusterMetadata:
+    # Mutable until installed=true, after which we will not allow updates.
+    infraID: ci-cluster-v4-1-64bns
+    clusterID: 1e4f00a3-a30e-46d7-9567-4bb61caf6d73
+    adminKubeconfigSecret:
+      name: ci-cluster-v4-1-0-ndv7l-admin-kubeconfig
+    adminPasswordSecret:
+      name: ci-cluster-v4-1-0-ndv7l-admin-password
+  installed: true
+  manageDNS: true # immutable
+  platform:
+    aws:
+      region: us-east-1
+  platformSecrets:
+    aws:
+      credentials:
+        name: aws
+  pullSecret:
+    name: pull
+status:
+  apiURL: https://api.ci-cluster-v4-1.r4f1.s1.devshift.org:6443
+  certificateBundles:
+  - generated: true
+    name: primary-cert-bundle
+  cliImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:528f2ead3d1605bdf818579976d97df5dd86df0a2a5d80df9aa8209c82333a86
+  clusterVersionStatus:
+    [SNIP]
+  conditions:
+  - lastProbeTime: "2019-10-23T08:40:24Z"
+    lastTransitionTime: "2019-10-23T08:40:24Z"
+    message: Control plane certificates are present
+    reason: ControlPlaneCertificatesFound
+  installedTimestamp: "2019-10-23T08:38:03Z"
+  installerImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:c9c1e6d1897b74d366d331f5cdc3fa16c5a96abe28ff8238833c09042156e983
+  provision:
+    name: ci-cluster-v4-1-0-ndv7l
+  webConsoleURL: https://console-openshift-console.apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+```
+
+## ClusterProvision
+
+ 1. Drop PodSpec, generate on the fly? (note: this is not within the Spec, external, possibly already generated on fly)
+
+## ClusterDeprovisionRequest
+
+ 1. Rename to ClusterDeprovision, to better match ClusterProvision.
+
+## SyncSet + SelectorSyncSet
+
+ 1. Drop support for ApplyOnce patch mode.
+
+## HiveConfig
+
+ 1. Adjust HiveConfig API to map managed DNS domains to cloud provider they should be used for.
+   * Would allow using Hive to provision to multiple clouds, using that cloud for DNS forwarding, from one Hive.
+   * This is not the expected deployment model in Dedicated (one Hive per cloud)
+   * Convert managedDomains to a list of structs rather than just a string. Struct should include a list of the platforms we will manage DNS for on this sub-domain.
+ 1. Rename ExternalDNS to ManageDNS (section where the credentials are stored, ExternalDNS being phased out)
+
+
+# Migrating to V1
+
+CRD versioning functionality is not available in OpenShift 3.11 where prod/stage run and will continue to do so until private cluster support is ready and we can migrate. We must however move on V1 now.
+
+Possible solutions:
+
+  1. Fork the codebase. Branch for dedicated and backport critical fixes only. Once migrated to 4.x they can upgrade to newer Hive and gain access to the V1 API.
+  2. Introduce a new parallel APIGroup.
```

Comment:
@csrwng is this what you had in mind or are there other options I'm not thinking of?

Reply:
Yes, either that or come up with new type names